### PR TITLE
issue: 1512054 Enable direct mlx5 processing on VMs

### PR DESCRIPTION
**Pull request by liranoz12**
mlx5 qp will be created only when bf could be used.

Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by liranoz12** [comment]
#717 - should resolve the limitation to create mlx5 qp even when bf could not be used.

File: src/vma/dev/qp_mgr.h [hunk 1]
```diff
@@ -215,6 +215,24 @@ class qp_mgr_eth : public qp_mgr
 			throw_vma_exception("failed creating qp");
 	};
 
+	static bool     is_mlx5_qp_supported(const ib_ctx_handler* ib_ctx) {
+		#define VMA_MLX5_MMAP_GET_WC_PAGES_CMD  2
+		#define VMA_MLX5_IB_MMAP_CMD_SHIFT      8
+
+		if (strstr(((ib_ctx_handler*)ib_ctx)->get_ibname(), "mlx5")) {
```
Comment (rafiw) [inline src/vma/dev/qp_mgr.h]: https://github.com/Mellanox/libvma/blob/6416e3e9e0ecf39d1c96addbdc3008105a16d7f4/src/vma/dev/qp_mgr.h#L144

Comment (liranoz12) [reply src/vma/dev/qp_mgr.h]: I do not understand what that means,
I am planning to remove this function because no one is using it.

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
TODO : add \n in the debug prints

**Comment by liranoz12** [comment]
duplicated by #717

**Pull request closed without merging**